core/sequencer/sequencer.go

```diff
@@ -27,42 +27,49 @@ import (
 
 	"github.com/golang/glog"
 	"github.com/golang/protobuf/proto"
-	"github.com/prometheus/client_golang/prometheus"
+	"github.com/google/trillian/monitoring"
 
 	pb "github.com/google/keytransparency/core/api/v1/keytransparency_go_proto"
 	tpb "github.com/google/trillian"
 	tclient "github.com/google/trillian/client"
 )
 
-var (
-	mutationsCTR = prometheus.NewCounter(prometheus.CounterOpts{
-		Name: "kt_signer_mutations",
-		Help: "Number of mutations the signer has processed.",
-	})
-	indexCTR = prometheus.NewCounter(prometheus.CounterOpts{
-		Name: "kt_signer_mutations_unique",
-		Help: "Number of mutations the signer has processed post per epoch dedupe.",
-	})
-	mapUpdateHist = prometheus.NewHistogram(prometheus.HistogramOpts{
-		Name:    "kt_signer_map_update_seconds",
-		Help:    "Seconds waiting for map update",
-		Buckets: []float64{0.005, 0.01, 0.025, 0.05, 0.1, 0.25, 0.5, 1, 2.5, 5, 10, math.Inf(1)},
-	})
-	createEpochHist = prometheus.NewHistogram(prometheus.HistogramOpts{
-		Name:    "kt_signer_create_epoch_seconds",
-		Help:    "Seconds spent generating epoch",
-		Buckets: []float64{0.005, 0.01, 0.025, 0.05, 0.1, 0.25, 0.5, 1, 2.5, 5, 10, math.Inf(1)},
-	})
+const (
+	// MaxBatchSize limits the number of mutations that will be processed per epoch.
+	MaxBatchSize  = int32(1000)
+	domainIDLabel = "domainid"
 )
 
-// MaxBatchSize limits the number of mutations that will be processed per epoch.
-const MaxBatchSize = int32(1000)
+var (
+	once               sync.Once
+	knownDomains       monitoring.Gauge
+	mutationsCTR       monitoring.Counter
+	sequencingRuns     monitoring.Counter
+	sequencingFailures monitoring.Counter
+	sequencingLatency  monitoring.Histogram
+)
 
-func init() {
-	prometheus.MustRegister(mutationsCTR)
-	prometheus.MustRegister(indexCTR)
-	prometheus.MustRegister(mapUpdateHist)
-	prometheus.MustRegister(createEpochHist)
+func createMetrics(mf monitoring.MetricFactory) {
+	knownDomains = mf.NewGauge(
+		"known_domains",
+		"Set to 1 for known domains (whether this instance is master or not)",
+		domainIDLabel)
```

Comment:
Is the final parameter 'domainid' defining a field within known_domains over which separate counts will be made?

Reply:
That's right.  My rough understanding is that this will produce a metric which contains a map from domain_id to this 1/0 count.